clients/gtest/CMakeLists.txt

```diff
@@ -195,16 +195,6 @@ if(HIP_PLATFORM STREQUAL amd)
     target_link_libraries( hipblas_v2-test PRIVATE hip::${CUSTOM_TARGET} )
   endif( )
 
-  if( CMAKE_CXX_COMPILER MATCHES ".*/hipcc$" )
-# hip-clang needs specific flag to turn on pthread and m
-    target_link_libraries( hipblas-test PRIVATE -lpthread -lm )
-    target_link_libraries( hipblas_v2-test PRIVATE -lpthread -lm )
-
-    if(BUILD_ADDRESS_SANITIZER)
-      target_link_libraries( hipblas-test PRIVATE -fuse-ld=lld -lgfortran )
-      target_link_libraries( hipblas_v2-test PRIVATE -fuse-ld=lld -lgfortran )
-    endif()
-  endif()
```

Comment:
Is this duplicated and already set in another file?    Would think they may still be required for amdclang ....

Reply:
I was able to build locally with `./install.sh -cn --compiler=hipcc` with these changes. Testing out address sanitizer on different arch now and will see how CI goes

Reply:
Address sanitizer was broken before this commit which may fix it, seems to be broken in rocblas as well. I'll see if I can fix it in rocblas tomorrow and try again in hipblas, but guessing it isn't used by anyone since the option is broken in our build scripts.